src/cert.rs

```diff
@@ -87,18 +89,33 @@ impl CertContext {
         }
     }
 
-    /// Return DER-encoded X.509 certificate chain
+    /// Return DER-encoded X.509 certificate chain.
+    /// (1) exclude the root. (2) check leaf cert to determin to use HKLM engine or HKCU engine
```

Comment:
Typo: `determin`. Just for my understanding, why do we want to exclude the root cert from the chain?

Reply:
I guess to better support certificate cross-signing with intermediate certs?

Reply:
The common practice for TLS is to exclude the root in the sequence of certificates included in the handshake. Trying to reduce the byte length of the TLS handshake. The client should already have the root that it needs to trust.